libs/balances-and-allowances/src/hooks/usePersistBalancesAndAllowances.ts

```diff
@@ -0,0 +1,107 @@
+import { useSetAtom } from 'jotai/index'
+import { useEffect, useMemo } from 'react'
+
+import { GP_VAULT_RELAYER } from '@cowprotocol/common-const'
+import { getIsNativeToken } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { useMultipleContractSingleData } from '@cowprotocol/multicall'
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { SWRConfiguration } from 'swr'
+
+import { erc20Interface } from '../const'
+import { AllowancesState, allowancesState } from '../state/allowancesAtom'
+import { balancesAtom, BalancesState } from '../state/balancesAtom'
+
+const MULTICALL_OPTIONS = {}
+
+export interface PersistBalancesAndAllowancesParams {
+  account: string | undefined
+  chainId: SupportedChainId
+  tokenAddresses: string[]
+  balancesSwrConfig: SWRConfiguration
+  allowancesSwrConfig: SWRConfiguration
+  setLoadingState?: boolean
+}
+
+export function usePersistBalancesAndAllowances(params: PersistBalancesAndAllowancesParams) {
```

Comment:
The code is actually just extracted from `BalancesAndAllowancesUpdater`

Reply:
Makes sense that you refactored this logic so you can use it in the 2 updaters. I wonder if we should even have a sigle updater, and just pass the refresh (so any lib can tell this updater the refresh interval). 

If we do that, we might want to extract the native into its own updater, which sounds like a good idea, as its completly separated logic and clients of this lib could use it in isolation.